src/modules/variables/variable.js

```diff
@@ -0,0 +1,144 @@
+export class Variable {
+  #variableStore
+
+  /**
+   * Creates an instance of the Variable class.
+   *
+   * @constructor
+   * @param {VariableStore} options.variableStore - The VariableStore instance for registration.
+   * @param {string|null} options.name - The name of the variable.
+   * @param {*} options.defaultValue - The default value of the variable.
+   * @param {*} options.value - The current value of the variable.
+   * @param {string|null} options.type - The type of the variable (string, number, boolean).
+   * @param {boolean} options.hadStarted - A flag indicating whether the variable has started (used internally).
+   * @param {Function[]} options.valueChangedCallbacks - Array to store callbacks to be executed when the variable value changes.
+   */
+  constructor ({ variableStore }) {
+    this.name = null
+    this.defaultValue = null
+    this.value = null
+    this.type = null
+    this.hadStarted = false
+    this.valueChangedCallbacks = []
+    this.#variableStore = variableStore
+  }
+
+  getValue () {
+    return this.value
+  }
+
+  getdefaultValue () {
+    return this.defaultValue
+  }
+
+  /**
+   * Defines a new variable with the provided name, default value, and variable store.
+   * @static
+   * @param {string} name - The name of the variable.
+   * @param {*} defaultValue - The default value of the variable.
+   * @param {VariableStore} variableStore - The VariableStore instance for registration.
+   * @returns {Variable|null} - The created Variable instance or null if invalid parameters are provided.
+   */
+  static define (name, defaultValue, variableStore) {
+    if (!name || typeof name !== 'string') {
+      console.error('Empty or invalid name parameter provided.')
+      return null
+    }
+    if (name.startsWith('.') || name.endsWith('.')) {
+      console.error('Variable name starts or ends with a `.` which is not allowed: ' + name)
+      return null
+    }
+
+    let type
+    if (typeof defaultValue === 'string') {
+      type = 'string'
+    } else if (typeof defaultValue === 'number') {
+      type = 'number'
+    } else if (typeof defaultValue === 'boolean') {
+      type = 'boolean'
+    } else {
+      console.error('Only primitive types (string, number, boolean) are accepted as value')
```

Comment:
This can be actually simplified
```
const typeOfDefaultValue = typeof defaultValue;
if (typeOfDefaultValue !== 'string' || typeOfDefaultValue !== 'number' || typeOfDefaultValue !== 'boolean') 
console.error('Only primitive types (string, number, boolean) are accepted as value')
```

We can use this new variable `typeOfDefaultValue`

Reply:
Updated this.